homeassistant/components/group/climate.py

```diff
@@ -0,0 +1,396 @@
+        self._attr_preset_mode: str | None = None
+
+    @callback
+    def async_update_supported_features(
+        self,
+        entity_id: str,
+        new_state: State | None,
+    ) -> None:
+        """Update dictionaries with supported features."""
+        if not new_state:
+            for climate in self._features.values():
+                climate.discard(entity_id)
+            return
+
+        new_features = new_state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+        for feature_flag, entity_set in self._features.items():
+            if new_features & feature_flag:
+                entity_set.add(entity_id)
+            else:
+                entity_set.discard(entity_id)
+
+    @callback
+    def async_update_group_state(self) -> None:
+        """Query all members and determine the climate group state."""
+        self._attr_assumed_state = False
+
+        states = [
+            state
+            for entity_id in self._entity_ids
+            if (state := self.hass.states.get(entity_id)) is not None
+        ]
+        self._attr_assumed_state |= not states_equal(states)
+
+        # Set group as unavailable if all members are unavailable or missing
+        self._attr_available = any(state.state != STATE_UNAVAILABLE for state in states)
+
+        # Temperature settings
+        self._attr_target_temperature = reduce_attribute(
+            states, ATTR_TEMPERATURE, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_target_temperature_step = reduce_attribute(
+            states, ATTR_TARGET_TEMP_STEP, reduce=max
+        )
+
+        self._attr_target_temperature_low = reduce_attribute(
+            states, ATTR_TARGET_TEMP_LOW, reduce=lambda *data: mean(data)
+        )
+        self._attr_target_temperature_high = reduce_attribute(
+            states, ATTR_TARGET_TEMP_HIGH, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_current_temperature = reduce_attribute(
+            states, ATTR_CURRENT_TEMPERATURE, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_min_temp = reduce_attribute(states, ATTR_MIN_TEMP, reduce=max)
+        self._attr_max_temp = reduce_attribute(states, ATTR_MAX_TEMP, reduce=min)
+        # End temperature settings
+
+        # Build util that will help with the computation of union of modes
+        # across all the entity states.
+        def merge_modes(modes: list[list[Any]]) -> list[Any]:
+            return list(set().union(*modes))
+
+        # available HVAC modes
+        all_hvac_modes = list(find_state_attributes(states, ATTR_HVAC_MODES))
+        if all_hvac_modes:
+            self._attr_hvac_modes = merge_modes(all_hvac_modes)
+
+        current_hvac_modes = [
+            x.state
+            for x in states
+            if x.state not in [HVACMode.OFF, STATE_UNAVAILABLE, STATE_UNKNOWN]
+        ]
+        # return the most common hvac mode (what the thermostat is set to do) except OFF, UNKNOWN and UNAVAILABE
```

Comment:
On this one and remaining below. Should we really return the most common one instead of having a sort off priority of states?
Example: One is heating and two are off. Should we then not return `heat` instead of `off`?

Reply:
HVAC modes used as state of a climate entity can be seen as an binary state with multiple ON states. Here the logic is filtering out the negative stats (OFF, UNAVAILABLE) so if any entity state is active (or ON state) those values are taken into account. Then similarly on how some enums attributes are being aggregated across entities in other groups, taking the most common enum value has been the consensus at the moment.

Reply:
Ok. So perhaps I took a bad example so let's do another one.
3 entities, one is `heat` and two are `fan_only`. With your logic the group state would be `fan_only`.
Is that really what is wished here instead of returning `heat`?

Btw: Don't resolve until we agree, it's annoying to have to unresolve to continue the discussion

Reply:
> 3 entities, one is heat and two are fan_only. With your logic the group state would be fan_only.
Is that really what is wished here instead of returning heat?

Yes that was my intention since the beginning. I think the logic behind of why `heat` might be more important than `fan_only` could be driven that could be a more costly mode. This issue was raised on the discussion of the original  docs PR: https://github.com/home-assistant/home-assistant.io/pull/23990#discussion_r973765098

That being said, I believe that most common mode might be the fairest approach as ranking can be really hard to do. `heat` might be ranked higher than `cool` or `fan_only` on colder countries, while it could be the other way around on much more hot countries. This can be a matter of perspective depending on how costly each mode is on different climate environments, therefore having a more uniform approach looks the right solution to me by taking the most common mode across all working modes (excluding `off`).